### ISSUE TITLE
[google_maps_flutter] The app crashes when a screen with the GoogleMap widget is repeatedly navigated to

### ISSUE THREAD

**Issue by guilhermedaldim** [flutter/flutter#56442]
The app crashes after some accessing the screen with the GoogleMap widget, this occurs on devices and emulators. Take a look at the screenshot below:

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/42361211/81114218-f9f6f880-8ef7-11ea-9f30-ea2574bd4fe2.gif).

P.S.: In this snippet of test code, the app takes about 2 to 3 minutes to crash, the record is from the end of that period.

P.S. 2: In larger apps, the problem happens faster, it takes about 1 minute.

P.S. 3: I'm just waiting for this to publish the app.

The console does not generate an error, it just crashes:

![console](https://user-images.githubusercontent.com/42361211/81114824-1e070980-8ef9-11ea-8457-5c4d75afdd92.png)

Below the code used for the test:

```
import 'package:flutter/material.dart';
import 'package:google_maps_flutter/google_maps_flutter.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Map',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Go to Map'),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            FlatButton(
              child: Text('Go to Map', style: TextStyle(color: Colors.white)),
              color: Colors.blue,
              onPressed: () => Navigator.push(context, MaterialPageRoute(builder: (context) => Map())),
            ),
          ],
        ),
      ),
    );
  }
}

class Map extends StatefulWidget {
  @override
  _MapState createState() => _MapState();
}

class _MapState extends State<Map> {
  GoogleMapController mapController;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Map'),
      ),
      body: GoogleMap(
        compassEnabled: false,
        mapToolbarEnabled: false,
        zoomControlsEnabled: false,
        initialCameraPosition: CameraPosition(
          target: LatLng(37.42796133580664, -122.085749655962),
          zoom: 14.4746,
        ),
        onMapCreated: (GoogleMapController controller) {
          mapController = controller;       
        },
        markers: {
          Marker(
            markerId: MarkerId('Map'),
            position: LatLng(37.42796133580664, -122.085749655962),
            icon: BitmapDescriptor.defaultMarkerWithHue(BitmapDescriptor.hueRed),
          ),
        }
      ),
    );
  }
}
```

**Comment by VladyslavBondarenko** [comment]
Hi @guilhermedaldim 
what google_maps_flutter version do you use?
could you provide your `flutter doctor -v`?

With google_maps_flutter: 0.5.27+1 on `dev, 1.18.0-12.0.pre` on iPhone 7 iOS 13.4.5 I couldn't reproduce after about 100 opening-closing the map

**Comment by guilhermedaldim** [comment]
Hi @VladyslavBondarenko, thanks for your reply.

I'm using the latest version of google_maps_flutter: ```0.5.27 + 1```

I'm working on the beta,``` v1.17.0-3.4.pre```, but I tested it on all channels and found the crash on all.

Devices where I detected the problem:
- Pixel 3 XL Android 10 (API 29)  (Emulator), the crash occurs more quickly;
- Xiaomi Mi 8 Android 10 (API 29);
- Asus Zenfone Max Pro M1 Android 9 (API 28).

**Edit:**
I haven't tested on ios devices.

<details><summary>Flutter doctor -v:</summary>
<p>

```
[√] Flutter (Channel beta, v1.17.0-3.4.pre, on Microsoft Windows [versÃ£o 10.0.18363.778], locale pt-BR)
    • Flutter version 1.17.0-3.4.pre at C:\flutter
    • Framework revision e6b34c2b5c (5 days ago), 2020-05-02 11:39:18 -0700
    • Engine revision 540786dd51
    • Dart version 2.8.1        

 
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\Users\Guilherme\AppData\Local\Android\sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[√] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] VS Code (version 1.44.2)
    • VS Code at C:\Users\Guilherme\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.10.1

[√] Connected device (2 available)
    • ASUS X00TDB               • J9AXB601K930JN6 • android-arm64 • Android 9 (API 28)
    • Android SDK built for x86 • emulator-5554   • android-x86   • Android 10 (API 29) (emulator)

• No issues found!
```

</p>
</details>

**Comment by brizaldi** [comment]
this issue happens to me as well, tested on android device Xiaomi Redmi Note 7,
it's possible that theres a problem with allocated memory not disposed correctly, after opening and closing screen with GoogleMap my app memory increased to over 1GB and then the app crashed

tested on Flutter channel master 1.18.0-14.0.pre.19 and google_maps_flutter v0.5.27+1

**Comment by guilhermedaldim** [comment]
In some more tests, I could see that the crash starts to occurs in version ```0.5.26+4```.

The workaround solution, for now, would be to use google_maps_flutter in version ```0.5.26+3```.

**Comment by brizaldi** [comment]
> 
> 
> In some more tests, I could see that the crash starts to occurs in version `0.5.26+4`.
> 
> The workaround solution, for now, would be to use google_maps_flutter in version `0.5.26+3`.

the memory leak is still there even after using `v0.5.26+3` or any version older than that, it might not be noticeable on small app, but on more complex app the difference on used memory between before and after opening the page with map inside it is quite big

in my case just opening page with maps took at least 150mb of memory, and that memory is gradually getting bigger and never disposed

**Comment by flutter2k20** [comment]
We are also facing the exact same issue, the application view containing the map goes blank if the app goes to the background (e.g. by hitting the home button) and stays there for some time, bringing the app back to the foreground makes the map screen go blank.
Memory consumption is another issue, for a single screen with a map consumes at least 150mb, and it keeps on going, I have seen it go to at least 500 or so mb, for no reason.

We are on the lastest flutter stable channel, and also on the latest version of the google maps plugin, 0.5.27+3. We tested on Asus Zenfone 3 (Android 8) and Samsung S9 (Android 10), same behavior on both devices. Also, same behavior with debug or release build.

**Can someone please raise the priority of this issue to high? This is a blocker for us, and we can't release the app without this issue getting fixed!**

**Comment by rvanhee** [comment]
We've seen a major improvement to the Google Maps memory consumption on Android by applying this one line change to the Flutter SDK: 
https://github.com/flutter/flutter/pull/50111

**Comment by flutter2k20** [comment]
> We've seen a major improvement to the Google Maps memory consumption on Android by applying this one line change to the Flutter SDK:
> #50111

I have tried out that the fix, but unfortunately, it didn't help :(. I could still see the memory grow up to a half a gig and also the google maps crashing/going blank after some time.

**Comment by cyanglaz** [comment]
Thanks for raising this issue. This regression is caused by https://github.com/flutter/plugins/pull/2704
https://github.com/flutter/plugins/pull/2772 should fix it.

**Comment by cyanglaz** [comment]
https://github.com/flutter/plugins/pull/2772 has landed. The memory leak issue directly contributes to the crash mentioned in the issue should be fixed in 0.5.28+1

**Comment by guilhermedaldim** [comment]
> [flutter/plugins#2772](https://github.com/flutter/plugins/pull/2772) has landed. The memory leak issue directly contributes to the crash mentioned in the issue should be fixed in 0.5.28+1

This worked for me, thanks for the solution @cyanglaz.

I tested it in the test application mentioned above and in a more complex application, and in both, the failure no longer occurs.

**Comment by cyanglaz** [comment]
@guilhermedaldim Thank you for verifying. I'm closing the issue.
